source/lang/walk.d

```diff
@@ -281,111 +280,166 @@ class Walker
         pushInstr(func, Instr(mixin("Opcode.op" ~ op)));
     }
 
-    void walkSet(Node[] c)
+    void walkSetLeft(Node left)
     {
-        foreach (p; 0 .. c.length / 2)
-        {
-            walk(c[p * 2 + 1]);
-        }
-        foreach_reverse (p; 0 .. c.length / 2)
+        if (Call call = cast(Call) left)
         {
-            Node target = c[p * 2];
-            if (cast(Call) target)
+            Ident id = cast(Ident) call.args[0];
+            switch (id.repr)
             {
-                Call call = cast(Call) target;
-                Ident ident = cast(Ident) call.args[0];
-                string name = ident.repr;
-                switch (name)
+            case "@index":
+                walk(call.args[1]);
+                walk(call.args[2]);
+                break;
+            case "@array":
```

Comment:
In the future this could be a numeric enum? Don't really know enough about the source though yet.